agent/src/jmh/java/com/code_intelligence/jazzer/instrumentor/EdgeCoverageInstrumentation.java

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.instrumentor;
+
+import static java.lang.String.format;
+
+import com.code_intelligence.jazzer.MockDriver;
+import com.code_intelligence.jazzer.runtime.CoverageMap;
+import com.github.fmeum.rules_jni.RulesJni;
+import java.io.*;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.openjdk.jmh.annotations.*;
+
+@Warmup(iterations = 10, time = 3)
+@Measurement(iterations = 10, time = 3)
+@Fork(value = 3)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@BenchmarkMode(Mode.AverageTime)
+@State(Scope.Benchmark)
+@SuppressWarnings("unused")
+public class EdgeCoverageInstrumentation {
+  private Object exampleObject;
+  private Method exampleMethod;
+
+  static {
+    MockDriver.load();
+    RulesJni.loadLibrary("coverage_tracker_jni", "/driver");
+  }
+
+  @Setup
+  public void setupInstrumentation() throws Exception {
+    String outDir = System.getenv("TEST_UNDECLARED_OUTPUTS_DIR");
+    if (outDir == null || outDir.isEmpty()) {
+      outDir = File.createTempFile("tmp", "tmp").getParent();
+    }
+
+    byte[] originalBytecode = classToBytecode(ExampleClass.class);
+    dumpBytecode(outDir, ExampleClass.class.getName(), originalBytecode);
+
+    byte[] patchedBytecode = applyInstrumentation(originalBytecode);
+    dumpBytecode(outDir, ExampleClass.class.getName() + ".patched", patchedBytecode);
+
+    Class<?> patchedClass = bytecodeToClass(ExampleClass.class.getName(), patchedBytecode);
+    exampleObject = patchedClass.getDeclaredConstructor().newInstance();
+    exampleMethod = patchedClass.getDeclaredMethod("exampleMethod");
+  }
+
+  @Benchmark
+  public Object benchmarkInstrumentedMethodCall() throws Exception {
+    return exampleMethod.invoke(exampleObject);
+  }
+
+  public static class ExampleClass {
+    private final Random rnd = new Random();
+    public List<Integer> exampleMethod() {
+      // optimized calls
+      ArrayList<Integer> strings = new ArrayList<>();
+      strings.add(rnd.nextInt());
+      strings.add(rnd.nextInt());
+      strings.add(rnd.nextInt());
+      strings.add(rnd.nextInt());
+      strings.add(rnd.nextInt());
+      // multiple calls on one line
+      int i = rnd.nextInt() + 10 * 10;
```

Comment:
I only see a single call on this line, what is the intention?

Reply:
The comment is outdated. Added another `rnd.nextInt()` call on that line and removed the comment.